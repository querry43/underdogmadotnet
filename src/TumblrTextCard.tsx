import * as React from 'react'

import TumblrCard from './TumblrCard'

class TumblrTextCard extends TumblrCard {
  public render() {
    return (
      <div>
        {this.props.post.type}
      </div>
    )
  }
}

export default TumblrTextCard

// {
//   "post": {
//     "type": "text",
//     "blog_name": "qrry43",
//     "blog": {},
//     "id": 179779160723,
//     "post_url": "https://qrry43.tumblr.com/post/179779160723/test-text",
//     "slug": "test-text",
//     "date": "2018-11-05 05:13:20 GMT",
//     "timestamp": 1541394800,
//     "state": "published",
//     "format": "html",
//     "reblog_key": "oZOxjg3Z",
//     "tags": {},
//     "short_url": "https://tmblr.co/ZlGD6Y2dRh4gJ",
//     "summary": "test text",
//     "is_blocks_post_format": false,
//     "recommended_source": null,
//     "recommended_color": null,
//     "note_count": 0,
//     "title": "test text",
//     "body": "<p>\n\ntest text\n\n<br/></p>",
//     "reblog": {},
//     "trail": {},
//     "can_like": false,
//     "can_reblog": false,
//     "can_send_in_message": true,
//     "can_reply": false,
//     "display_avatar": true
//   }
// }
