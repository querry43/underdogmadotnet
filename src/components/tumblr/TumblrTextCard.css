.TumblrTextCard_body > ul {
  text-align: left;
}
