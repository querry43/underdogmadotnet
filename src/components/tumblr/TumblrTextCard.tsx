import * as React from 'react'

import TumblrCard from './TumblrCard'
import './TumblrTextCard.css'

const titleStyle : React.CSSProperties = {
  color: 'black',
  textDecoration: 'none'
}

class TumblrTextCard extends TumblrCard {
  protected content() {
    return (
      <div className="TumblrTextCard">
        <p style={titleStyle}>{this.props.post.title}</p>
        <div className="TumblrTextCard_body" dangerouslySetInnerHTML={{ __html: this.props.post.body }} />
      </div>
    )
  }
}

export default TumblrTextCard
