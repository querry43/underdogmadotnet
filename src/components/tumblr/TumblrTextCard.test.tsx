/* tslint:disable:object-literal-sort-keys */

import * as React from 'react'

import TumblrTextCard from './TumblrTextCard'

it('matches snapshot', () => {
  const post = {
    "type": "text",
    "blog_name": "qrry43",
    "blog": {},
    "id": 179779160723,
    "post_url": "https://qrry43.tumblr.com/post/179779160723/test-text",
    "slug": "test-text",
    "date": "2018-11-05 05:13:20 GMT",
    "timestamp": 1541394800,
    "state": "published",
    "format": "html",
    "reblog_key": "oZOxjg3Z",
    "tags": {},
    "short_url": "https://tmblr.co/ZlGD6Y2dRh4gJ",
    "summary": "test text",
    "is_blocks_post_format": false,
    "recommended_source": null,
    "recommended_color": null,
    "note_count": 0,
    "title": "test text",
    "body": "<p>\n\ntest text\n\n<br/></p>",
    "reblog": {},
    "trail": {},
    "can_like": false,
    "can_reblog": false,
    "can_send_in_message": true,
    "can_reply": false,
    "display_avatar": true
  }

  expect(<TumblrTextCard post={post} />).toMatchSnapshot()
)
