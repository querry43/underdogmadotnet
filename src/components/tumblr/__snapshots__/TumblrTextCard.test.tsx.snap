// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<TumblrTextCard
  post={
    Object {
      "blog": Object {},
      "blog_name": "qrry43",
      "body": "<p>

test text

<br/></p>",
      "can_like": false,
      "can_reblog": false,
      "can_reply": false,
      "can_send_in_message": true,
      "date": "2018-11-05 05:13:20 GMT",
      "display_avatar": true,
      "format": "html",
      "id": 179779160723,
      "is_blocks_post_format": false,
      "note_count": 0,
      "post_url": "https://qrry43.tumblr.com/post/179779160723/test-text",
      "reblog": Object {},
      "reblog_key": "oZOxjg3Z",
      "recommended_color": null,
      "recommended_source": null,
      "short_url": "https://tmblr.co/ZlGD6Y2dRh4gJ",
      "slug": "test-text",
      "state": "published",
      "summary": "test text",
      "tags": Object {},
      "timestamp": 1541394800,
      "title": "test text",
      "trail": Object {},
      "type": "text",
    }
  }
/>
`;
